geps/gep-2643/index.md

```diff
@@ -0,0 +1,263 @@
+  listeners:
+  - name: somelistener
+    port: 443
+    protocol: TLS
+    hostname: "*.example.com"
+    allowedRoutes:
+      namespaces:
+        from: Same
+      kinds:
+      - kind: TLSRoute
+    tls:
+      mode: Passthrough
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: TLSRoute
+metadata:
+  name: my-tls-route
+spec:
+  parentRefs:
+  - name: gateway-tlsroute
+  hostnames:
+  - abc.example.com
+  rules:
+  - backendRefs:
+    - name: tls-backend
+      port: 443
+```
+
+A typical [north/south](https://gateway-api.sigs.k8s.io/concepts/glossary/#northsouth-traffic)
+API request flow for a gateway implemented using a `TLSRoute` is:
+* A client makes a request to https://foo.example.com.
+* DNS resolves the name to a `Gateway` address.
+* The reverse proxy receives the request on a `Listener` and uses the
+[Server Name Indication](https://datatracker.ietf.org/doc/html/rfc6066#section-3)
+attribute to match an `TLSRoute`.
+* The reverse proxy passes through the request directly to one or more objects,
+i.e. `Service`, in the cluster based on `backendRefs` rules of the `TLSRoute`.
+
+## Conflict management and precedences
+
+The following conflict situations are covered by TLSRoute and TLS passthrough cases:
+
+* When a Gateway contains a listener with `protocol=TLS`, the Gateway MUST NOT
+allow any other kind of listener on the same port and the Gateway SHOULD be
+marked as `Accepted=False` in case there are different kinds of listeners on
+the same port.  
+* When a Gateway contains a listener with `protocol=TLS` and `tls.mode=Passthrough`,
+the `Gateway` MUST NOT allow another listener on the same port with a different
+`tls.mode` and the `Gateway` SHOULD be marked as `Accepted=False`.  
+* Any violating Listener should have a Condition `Conflicted=True`.  
+* If a hostname is specified by both the `Listener` and `TLSRoute`, there must
+be at least one intersecting hostname for the `TLSRoute` to be attached to the
+`Listener`.  
+  * A `Gateway listener` with `test.example.com` as the hostname matches a `TLSRoute` that
+  specifies `test.example.com` but does not match a `TLSRoute` that specifies `*.example.com`  
+  * A `Gateway listener` with `*.example.com` as the hostname matches a `TLSRoute` that
+  specifies at least one hostname that matches the `Gateway listener` hostname.
+  For example, `test.example.com` and `*.example.com` would both match. On the
+  other hand, `example.com` and `test.example.net` would not match.  
+  * If both the `Gateway listener` and `TLSRoute` specify hostnames, any `TLSRoute`
+  hostnames that do not match the `Gateway listener` hostname MUST be ignored
+  for that Listener. For example, if a `Gateway listener` specified `*.example.com`,
+  and the  `TLSRoute` specified `test.example.com` and `test.example.net`,
+  the later must not be considered for a match.  
+  * In any of the cases above, the `TLSRoute` should have a `Condition` of `Accepted=True`.
+
+## Multiplexing support
```

Comment:
This part needs some discussion/attention:
* Is Multiplexing support a core feature, or a implementation specific feature?
* Previously we state that a Gateway with a TLS termination can only have TLSRoutes, but here we say that multiple listeners on the same port, for different types can be accepted. So the conditions and conflict management should be changed to reflect this (and the conformance tests), if we agree that this case is possible
* What conditions should an implementation add when this is not supported? We need to word it explicitly on the GEP and the expected conformance tests

Reply:
I think we can add initially the multiplexing as implementation specific as is. 

I did a test with Istio and was able to check that it works OOTB, I expect that at least any other envoy, haproxy and nginx implementation can work with it.